draft-yasskin-httpbis-origin-signed-exchanges-impl.md

```diff
@@ -0,0 +1,807 @@
+
+{{?I-D.ietf-httpbis-header-structure}} provides a way to parameterise labels but
+not other supported types like binary content. If the `Signature` header field
+is notionally a list of parameterised signatures, maybe we should add a
+"parameterised binary content" type.
+{:#parameterised-binary}
+
+Should the certUrl and validityUrl be lists so that intermediates can offer a
+cache without losing the original URLs? Putting lists in dictionary fields is
+more complex than {{?I-D.ietf-httpbis-header-structure}} allows, so they're
+single items for now.
+
+## CBOR representation of exchange headers ## {#cbor-representation}
+
+To sign an exchange's headers, they need to be serialized into a byte string.
+Since intermediaries and distributors might
+rearrange, add, or just reserialize headers, we can't use the literal bytes of
+the headers as this serialization. Instead, this section defines a CBOR
+representation that can be embedded into other CBOR, canonically serialized
+({{canonical-cbor}}), and then signed.
+
+The CBOR representation of an exchange `exchange`'s headers is the CBOR
+({{!RFC7049}}) array with the following content:
+
+1. The map mapping:
+   * The byte string ':method' to the byte string containing `exchange`'s
+     request's method.
+   * The byte string ':url' to the byte string containing `exchange`'s request's
+     effective request URI.
+   * For each request header field in `exchange`, the header field's name as a
+     byte string to the header field's value as a byte string.
+1. The map mapping:
+   * the byte string ':status' to the byte string containing `exchange`'s
+     response's 3-digit status code, and
+   * for each response header field in `exchange`, the header field's name as a
+     byte string to the header field's value as a byte string.
+
+### Example ### {#example-cbor-representation}
+
+Given the HTTP exchange:
+
+~~~http
+GET https://example.com/ HTTP/1.1
+Accept: */*
+
+HTTP/1.1 200
+Content-Type: text/html
+Content-Encoding: mi-sha256
+MI: mi-sha256=20addcf7368837f616d549f035bf6784ea6d4bf4817a3736cd2fc7a763897fe3
+
+<0x0000000000004000><!doctype html>
+<html>
+...
+~~~
+
+The cbor representation consists of the following item, represented using the
+extended diagnostic notation from {{?I-D.ietf-cbor-cddl}} appendix G:
+
+~~~cbor-diag
+[
+  {
+    ':url': 'https://example.com/'
+    ':method': 'GET',
+  },
+  {
+    'mi': 'mi-sha256=20addcf7368837f616d549f035bf6784ea6d4bf4817a3736cd2fc7a763897fe3',
+    ':status': '200',
+    'content-type': 'text/html'
+    'content-encoding': 'mi-sha256',
+  }
+]
+~~~
+
+## Loading a certificate chain ## {#cert-chain-format}
+
+The resource at a signature's `certUrl` MUST have the
+`application/tls-cert-chain` content type and MUST contain a TLS 1.3 Certificate
+message (Section 4.4.2 of {{!I-D.ietf-tls-tls13}}) containing X.509v3
+certificates.
+
+Parsing notes:
+1. This resource MUST NOT include the 4-byte header that would appear in a
```

Comment:
Somehow this formatting is broken on the preview HTML version

Reply:
Fixed. The parser expects a blank line before lists.